pkg/clusteragent/autoscaling/controller.go

```diff
@@ -54,8 +54,14 @@ func NewController(
 		Client:    client,
 		Lister:    mainInformer.Lister(),
 		synced:    mainInformer.Informer().HasSynced,
-		Workqueue: workqueue.NewRateLimitingQueue(workqueue.DefaultItemBasedRateLimiter()),
-		IsLeader:  isLeader,
+		Workqueue: workqueue.NewRateLimitingQueueWithConfig(
```

Comment:
Controller is generic (even though in `autoscaling` currently, we should probably pass the workqueue as parameter.